agent/xds/clusters.go

```diff
@@ -176,6 +168,58 @@ func makeExposeClusterName(destinationPort int) string {
 	return fmt.Sprintf("exposed_cluster_%d", destinationPort)
 }
 
+// In transparent proxy mode there are potentially multiple passthrough clusters added.
+// The first is for destinations outside of Consul's catalog. This is for a plain TCP proxy.
+// All of these use Envoy's ORIGINAL_DST listener filter, which forwards to the original
+// destination address (before the iptables redirection).
+// The rest are for destinations inside the mesh, which require certificates for mTLS.
+func makePassthroughClusters(cfgSnap *proxycfg.ConfigSnapshot) ([]proto.Message, error) {
+	// This size is a lower bound.
+	clusters := make([]proto.Message, 0, len(cfgSnap.ConnectProxy.PassthroughUpstreams))
+
+	if cfgSnap.ConnectProxy.MeshConfig == nil ||
+		!cfgSnap.ConnectProxy.MeshConfig.TransparentProxy.CatalogDestinationsOnly {
+
+		clusters = append(clusters, &envoy_cluster_v3.Cluster{
+			Name: OriginalDestinationClusterName,
+			ClusterDiscoveryType: &envoy_cluster_v3.Cluster_Type{
+				Type: envoy_cluster_v3.Cluster_ORIGINAL_DST,
+			},
+			LbPolicy:       envoy_cluster_v3.Cluster_CLUSTER_PROVIDED,
+			ConnectTimeout: ptypes.DurationProto(5 * time.Second),
+		})
+	}
+
+	for _, passthrough := range cfgSnap.ConnectProxy.PassthroughUpstreams {
+		// Prefixed with passthrough to distinguish from non-passthrough clusters for the same upstream.
+		name := "passthrough." + passthrough.SNI
```

Comment:
What if someone had a subset of a cluster named `passthrough`? I think then you would have two clusters with the same name potentially generated and they'd conflict with each other (the tproxy passthrough, and the plain subset)

Reply:
You might go with the something like the tilde trick I used for the "customized hash" prefixes.

Reply:
Ah yep, swapping this out for a tilde as well.